### PR TITLE
gokrazy: init at unstable-2023-08-12

### PR DESCRIPTION
**Pull request by shayne**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 251277` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gok</li>
  </ul>
</details>

**Comment by nagy** [comment]
I think the attribute name of this should be `gokrazy` because `gok` is only the (short) name of the binary and that can be changed with `mainProgram = "gok";` in the meta attribute.

File: pkgs/development/misc/gok/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+buildGoModule rec {
+  pname = "gok";
+  version = "unstable-2023-08-12";
+
+  src = fetchFromGitHub {
+    owner = "gokrazy";
+    repo = "tools";
+    rev = "23cde3b0d858497a63c21e93ad30859bf197995f";
+    sha256 = "sha256-oqtkC04TaOkcXkGAZzATCBA0XnFsx7bSGP9ODyhgAxQ=";
```
Comment (nagy) [inline pkgs/development/misc/gok/default.nix:11]: can you use `hash` ?

File: pkgs/development/misc/gok/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  vendorHash = "sha256-rIIMqYMgLNCMYEH+44v79i8yGbHDmUY21X3h1E2jP9Q=";
+
+  ldflags = [ "-s" "-w" "-X=main.Version=${version}" ];
+
+  subPackages = [ "cmd/gok" ];
+
+  meta = with lib; {
+    description = "Turn your Go program(s) into an appliance running on the Raspberry Pi 3, Pi 4, Pi Zero 2 W, or amd64 PCs!";
+    homepage = "https://github.com/gokrazy/gokrazy";
+    license = licenses.bsd3;
```
[the remark this answers is not in the document]
Comment (shayne) [reply pkgs/development/misc/gok/default.nix]: Sure! Done.

File: pkgs/development/misc/gok/default.nix (hunk 1 above)
Comment (shayne) [reply pkgs/development/misc/gok/default.nix]: Done.

**Comment by nagy** [comment]
Can you separate your addition to the maintainer list into an own commit? after that, its fine.

**Comment by shayne** [comment]
OK. Split it up with the maintainers' commit being first.

**Review by nagy — APPROVED** [review]

**Review by marsam — APPROVED** [review]

**Pull request merged**